pkg/controller/upload-controller.go

```diff
@@ -623,7 +629,7 @@ func createUploadServiceNameFromPvcName(pvc string) string {
 	return naming.GetServiceNameFromResourceName(createUploadResourceName(pvc))
 }
 
-func (r *UploadReconciler) makeUploadPodSpec(args UploadPodArgs, resourceRequirements *v1.ResourceRequirements) *v1.Pod {
+func (r *UploadReconciler) makeUploadPodSpec(args UploadPodArgs, resourceRequirements *v1.ResourceRequirements, workloadNodePlacement cdiv1.NodePlacement) *v1.Pod {
```

Comment:
Any particular reason you are dereferencing the workloadNodePlacement above? Instead of passing a pointer as a parameter you are now passing the entire struct. I think you can have workloadNodePlacement *cdiv1.NodePlacement without any code changes in the actual function.